readme.md

```diff
@@ -1,848 +1,1214 @@
+		bar: 'bar'
+	};
 
-    // Literal notations:
-    var array = [],
-        object = {};
+```
 
+##### 2.2.2.2. One variable per scope
 
-    // 2.B.1.2
-    // Using only one `var` per scope (function) promotes readability
-    // and keeps your declaration list free of clutter (also saves a few keystrokes).
+Using only one variable per scope (function) promotes readability and keeps your declaration list free of clutter (also saves a few keystrokes). However, if a scope have [early returns](281-early-returns) statements, it will be more readable and performance wise declaring variable later, when definition is needed.
 
-        // Comment on this.
-    var foo = '',
-        bar = '',
-        // Comment on this.
-        quux;
+```javascript
+// Bad.
+function foo() {
+	// 2 statements
 
-    // 2.B.1.3
-    // Try to group `var` statements. Preferably in the beginning
-    // of their respective scope (function), but also don't go crazy with this.
-    // Avoid declaring new variable every few lines of code.
-    // In case of long functions make sure to declare all common variables
-    // (like loop counters, `el`, `range`, etc.) in the beginning of scope.
-    // Same goes for const and let from ECMAScript 6.
+	var bar = '';
 
-    // Bad.
-    function foo() {
-        // 2 statements
+	// 2 statements
 
-        var bar = '';
+	var foo = '';
 
-        // 2 statements
+	// 10 statements
 
-        var foo = '';
+	var bom = '';
 
-        // 50 statements
+	// some statements
+}
 
-        var bom = '';
+// Good
+function foo() {
+	var bar = '',
+		foo = '';
 
-        // some statements
-    }
+	// some statements
+}
 
-    // Good
-    function foo() {
-        var bar = '',
-            foo = '';
+// Bad
+function bar( x ) {
+	var foo = doSomethingSlow();
 
-        // 50 statements
+	if ( x < 2 ) {
+		return;
+	}
 
-        var bom = '';   // Assuming that bom isn't needed before, there's no sense
-                        // in grouping it with the first group.
+	// some statements
+}
 
-        // some statements
-    }
+// Good
+function bar( x ) {
+	if ( x < 2 ) {
+		return;
+	}
 
+	var foo = doSomethingSlow();
 
-    // 2.B.2.1
-    // Named Function Declaration
-    function foo( arg1, argN ) {
+	// some statements
+}
+```
 
-    }
+##### 2.2.2.3. Function declaration
 
-    // Usage.
-    foo( arg1, argN );
+Prefer function declaration instead of function expression to avoid issues with executing function before it has been defined.
 
+```javascript
+// Good
+function square( number ) {
+	return number * number;
+}
 
-    // 2.B.2.2
-    // Named Function Declaration.
-    function square( number ) {
-        return number * number;
-    }
+square( 10 );
 
-    // Usage.
-    square( 10 );
+// Bad
+var square = function( number ) {
+	return number * number;
+};
 
-    // Really contrived continuation passing style.
-    function square( number, callback ) {
-        callback( number * number );
-    }
+square( 10 );
+```
```

Comment:
Shouldn't be here also visible case related to: "to avoid issues with executing function before it has been defined"
So code should look like:
```javascript
// Good
square( 10 );

function square( number ) {
	return number * number;
}


// Bad
square( 10 );

var square = function( number ) {
	return number * number;
};
```